modules/packages/Crypto.chpl

```diff
@@ -0,0 +1,953 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*  A cryptographic library based on OpenSSl, targeted at flexible encryption
+    purposes.
+
+    The Crypto module focuses on providing various cryptographic utilities such as
+
+    - Symmetric Ciphers
+    - Asymmetric Ciphers
+    - Hashing Functions
+    - Key Derivation Functions
+    - Cryptographically-Secure Pseudo-Random Number Generators
+    - Miscellaneous Utilities such as CryptoBuffer, Envelopes, etc.
+
```

Comment:
It would be super useful to include a really simple hello-world style example in the documentation. Not necessary, but I think it will help users learn how to use this module quicker.